chain/chain/src/resharding.rs

```diff
@@ -107,24 +121,34 @@ impl Chain {
     fn build_state_for_split_shards_impl(
         state_split_request: StateSplitRequest,
     ) -> Result<HashMap<ShardUId, StateRoot>, Error> {
-        let StateSplitRequest {
-            runtime_adapter,
-            shard_uid,
-            state_root,
-            next_epoch_shard_layout,
-            ..
-        } = state_split_request;
-        // TODO(resharding) use flat storage to split the trie here
-        let tries = runtime_adapter.get_tries();
+        let StateSplitRequest { tries, shard_uid, state_root, next_epoch_shard_layout, .. } =
+            state_split_request;
+
+        let store = tries.get_store();
         let trie = tries.get_view_trie_for_shard(shard_uid, state_root);
+        let mut iter = store_helper::iter_flat_state_entries(shard_uid, &store, None, None).map(
+            |entry| -> (Vec<u8>, Vec<u8>) {
+                let (key, value) = entry.unwrap();
+                let value = match value {
+                    FlatStateValue::Ref(ref_value) => {
+                        trie.storage.retrieve_raw_bytes(&ref_value.hash).unwrap().to_vec()
+                    }
+                    FlatStateValue::Inlined(inline_value) => inline_value,
+                };
+                (key, value)
+            },
```

Comment:
nit: would be nice to move it to a helper method

Reply:
The best I can do is have a helper lambda method here as this function uses the captured variables `trie`. Would be difficult to abstract out as helper function.

Reply:
Actually, managed to shift this to a helper...